diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/WhenMustHaveElseWarnTest.kt

```diff
@@ -73,4 +73,21 @@ class WhenMustHaveElseWarnTest : LintTestBase(::WhenMustHaveElseRule) {
                 """.trimMargin()
         )
     }
+
+    @Test
+    @Tag(WarningNames.WHEN_WITHOUT_ELSE)
+    fun `regression - shouldn't check when in when branches and assignments`() {
+        lintMethod(
+            """
+                    |fun foo() {
+                    |    var x: Int
+                    |    x = when(it) {
+                    |        1 -> when (x) {
+                    |            2 -> foo()
+                    |        }
+                    |    }
```

Comment:
2 else should be here? why no warnings?

Reply:
Because in this cases return value of `when` expression is used afterwards, and the compiler can check whether the choice is exhaustive. Since we don't have information about expression type, we don't check these cases at all.